fedimint-client/src/meta.rs

```diff
@@ -0,0 +1,262 @@
+use std::collections::BTreeMap;
+use std::sync::Arc;
+use std::time::Duration;
+
+use anyhow::{bail, Context as _};
+use fedimint_core::db::{Database, IDatabaseTransactionOpsCoreTyped};
+use fedimint_core::task::waiter::Waiter;
+use fedimint_core::task::{MaybeSend, MaybeSync};
+use fedimint_core::util::{retry, FibonacciBackoff};
+use fedimint_core::{apply, async_trait_maybe_send};
+use serde::de::DeserializeOwned;
+use tracing::{debug, instrument, warn};
+
+use crate::db::{MetaFieldKey, MetaFieldPrefix, MetaFieldValue};
+use crate::Client;
+
+#[apply(async_trait_maybe_send!)]
+pub trait MetaSource: MaybeSend + MaybeSync + 'static {
+    /// Wait for next change in this source.
+    async fn wait_for_update(&self);
+    async fn fetch(
```

Comment:
`fetch`, should take `existing_revision: Option<u64>`, so in case of meta module, it can issue `check-revision` and 99.99% avoid downloading everything again.

Reply:
:+1: but can be done in 0.4 if it's a big concern. Your typical meta for one federation is rather small. To really implement it I'd like to add the previous revision to `GET_CONSENSUS_ENDPOINT`, so we don't need to make two calls but the server can decide if it should send us anything.

Reply:
done

Reply:
> Your typical meta for one federation is rather small.

The limit is 1MB.

Did you see the legal disclaimer about joining the pop-up federation? :D . That thin along makes me think maybe I should have made it 16MB.

I also expect people will want some kind of images sometimes.

And since you really want to continuously poll it, and it is supposed to work in rural African communities, probably worth it already, at least for some people.